ratis-common/src/main/java/org/apache/ratis/rpc/CallId.java

```diff
@@ -27,6 +28,17 @@
 public final class CallId {
   private static final AtomicLong CALL_ID_COUNTER = new AtomicLong();
 
+  private static final Comparator<Long> COMPARATOR = (left, right) -> {
+    final long diff = left - right;
+    // check diff < Long.MAX_VALUE/2 for the possibility of numerical overflow.
+    return diff == 0? 0: diff > 0 && diff < Long.MAX_VALUE/2? 1: -1;
```

Comment:
Why this `COMPARATOR` is different from GrpcLogAppender#CALL_ID_COMPARATOR?

Reply:
It generates call IDs in a different way -- this only allows >=0 values but GrpcLogAppender allows +/- values.

Reply:
Got it, thanks!